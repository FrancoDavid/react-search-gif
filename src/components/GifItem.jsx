import React from 'react';
import PropTypes from 'prop-types';

const GifItem = ({image}) => {
    return (
        <><img src={image.gif} alt={image.title} /></>
    );
}

GifItem.prototype = {
    image: {
        title: PropTypes.string.isRequired,
        gif: PropTypes.string.isRequired
    }
}
 
export default GifItem;