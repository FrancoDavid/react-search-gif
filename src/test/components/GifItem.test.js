import { shallow } from "enzyme";
import GifItem from "../../components/GifItem";
import React from 'react';

describe('Test Gif Item', () => {
    const obj_image = {
        title: 'Holamundo',
        gif: 'http://localhost'
    };
    const wrapper = shallow(<GifItem image={obj_image}/>);

    test('Show correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('props of image/gif', () => {
        const img = wrapper.find('img');

        expect(img.props().src).toBe(obj_image.gif);

        expect(img.props().alt).toBe(obj_image.title);
    });
});