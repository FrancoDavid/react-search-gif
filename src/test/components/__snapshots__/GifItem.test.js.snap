// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Gif Item Show correctly 1`] = `
<Fragment>
  <img
    alt="Holamundo"
    src="http://localhost"
  />
</Fragment>
`;
